src/FakeItEasy/Configuration/RuleBuilder.cs

```diff
@@ -143,12 +143,9 @@ public UnorderedCallAssertion MustHaveHappened(Repeated repeatConstraint)
 
             var asserter = this.asserterFactory.Invoke(this.Calls);
 
-            var description = new StringBuilderOutputWriter();
-            this.RuleBeingBuilt.WriteDescriptionOfValidCall(description);
+            asserter.AssertWasCalled(this.Matcher.Matches, this.RuleBeingBuilt.WriteDescriptionOfValidCall, repeatConstraint.Matches, repeatConstraint.ToString());

```

Comment:
Thanks, @thomaslevesque. Yeah, we could. I noticed the same thing, but figured it was _technically_ outside the mandate of the PR, so I didn't do it at that time, but it was a close thing. I'm happy to do now. 
I've added a commit (and rebased).